src/ralph/admin/tests/tests_filters.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+import datetime
+
+from django.test import TestCase
+
+from ralph.admin.filters import (
+    BooleanFilter,
+    ChoicesFilter,
+    DateFilter,
+    TextFilter,
+    date_format_to_human
+)
+from ralph.assets.models.choices import AssetStatus
+from ralph.data_center.admin import DataCenterAssetAdmin
+from ralph.data_center.models.physical import DataCenterAsset
+from ralph.data_center.tests.factories import DataCenterAssetFactory
+
+
+class AdminFiltersTestCase(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.dca_1 = DataCenterAssetFactory(
+            invoice_date=datetime.date(2015, 1, 1),
+            barcode='barcode_test',
+        )
+        cls.dca_2 = DataCenterAssetFactory(
+            invoice_date=datetime.date(2015, 2, 1)
+        )
+        cls.dca_2 = DataCenterAssetFactory(
+            invoice_date=datetime.date(2015, 3, 1),
+            force_depreciation=True
+        )
+
+    def test_date_format_to_human(self):
+        self.assertEqual('YYYY-MM-DD', date_format_to_human('%Y-%m-%d'))
+        self.assertEqual('YY-DD-MM', date_format_to_human('%y-%d-%m'))
+
+    def test_boolean_filter(self):
+        class ForceDepreciationFilter(BooleanFilter):
+            title = 'Force depreciation'
+            parameter_name = 'force_depreciation'
+
+        boolean_filter = ForceDepreciationFilter(
+            request=None,
+            params={'force_depreciation': True},
+            model=DataCenterAsset,
+            model_admin=DataCenterAssetAdmin,
+        )
+        queryset = boolean_filter.queryset(None, DataCenterAsset.objects.all())
+
+        self.assertEqual(1, queryset.count())
+
+    def test_choices_filter(self):
+        class StatusFilter(ChoicesFilter):
+            title = 'Status'
+            parameter_name = 'status'
+            choices_list = AssetStatus()
+
+        choices_filter = StatusFilter(
+            request=None,
+            params={'status': AssetStatus.new},
+            model=DataCenterAsset,
+            model_admin=DataCenterAssetAdmin,
+        )
+        queryset = choices_filter.queryset(None, DataCenterAsset.objects.all())
+
+        self.assertEqual(3, queryset.count())
+
+        choices_filter = StatusFilter(
+            request=None,
+            params={'status': AssetStatus.in_progress},
+            model=DataCenterAsset,
+            model_admin=DataCenterAssetAdmin,
+        )
+        queryset = choices_filter.queryset(None, DataCenterAsset.objects.all())
+
+        self.assertEqual(0, queryset.count())
+
+    def test_text_filter(self):
```

Comment:
please add one more test to search by value with separators (which will also test 'containing' - not full match)